### PR TITLE
Add Codecov support

### PR DESCRIPTION
**Pull request by masterleinad**
When running the testsuite witch checking for coverage also report to [Codecov](https://codecov.io/gh/dealii/dealii) which also allows for merging reports for a single commit and has a much nicer interface compared to the one provided by `CDash`.

### CODE REVIEW

File: tests/run_testsuite.cmake [hunk 1]
```diff
@@ -540,7 +547,10 @@ IF("${_res}" STREQUAL "0")
     IF(COVERAGE)
       CREATE_TARGETDIRECTORIES_TXT()
       MESSAGE("-- Running CTEST_COVERAGE()")
-      CTEST_COVERAGE()
+      #CTEST_COVERAGE()
+      FILE(DOWNLOAD "https://codecov.io/bash" "${CMAKE_CURRENT_BINARY_DIR}/tests/codecov-bash")
+      EXECUTE_PROCESS(COMMAND bash "${CMAKE_CURRENT_BINARY_DIR}/tests/codecov-bash"
+                              "-t ac85e7ce-5316-4bc1-a237-2fe724028c7b" "-x '${GCOV_COMMAND}'")
```
Comment (tamiko) [inline tests/run_testsuite.cmake]: Is there any other way than introducing a remote code execution in the `run_testsuite.cmake` script?

**Comment by tamiko** [comment]
Apart from the fact that I am adamant about not executing unsigned, remote controllable code, I think the dashboard provided by codecov is indeed much nicer.

File: tests/run_testsuite.cmake (hunk 1 above)
Comment (masterleinad) [reply tests/run_testsuite.cmake]: Not that I know of. The recommended (only?) way for uploading the coverage results is running this bash script.

Comment (tamiko) [reply tests/run_testsuite.cmake]: I am in favor of the latter approach - require to manually download and verify the script (and simply do not submit to codecov if the script is absent).

**Pull request merged**